File: org.palladiosimulator.blockchainsystems.core/src/main/java/org/palladiosimulator/blockchainsystems/core/transactionpropagation/TransactionReceivedTraceEvent.java
package org.palladiosimulator.blockchainsystems.core.transactionpropagation;

import org.palladiosimulator.blockchainsystems.core.common.abstractions.TraceEvent;
import org.palladiosimulator.blockchainsystems.core.system.abstractions.Transaction;
import org.palladiosimulator.blockchainsystems.core.system.abstractions.P2PNetworkEndpoint;

public record TransactionReceivedTraceEvent(
        long occurrenceTime,
        Transaction receivedTransaction,
        P2PNetworkEndpoint sendingEndpoint
) implements TraceEvent {

    public static final String EVENT_TYPE = "TransactionReceivedTraceEvent";

    @Override
    public long getOccurrenceTime() {
        return occurrenceTime;
    }

    public Transaction getSentTransaction() {
        return receivedTransaction;
    }

    public P2PNetworkEndpoint getReceivingNetworkEndpoint() {
        return sendingEndpoint;
    }

    @Override
    public String getEventType() {
        return EVENT_TYPE;
    }

    @Override
    public void formatDetails(StringBuilder stringBuilder) {
        stringBuilder.append("{ txId=");
        stringBuilder.append(receivedTransaction.getTxId());
        stringBuilder.append("; receivedFromNetworkEndpoint=");
        stringBuilder.append(sendingEndpoint.getEndpointId());
        stringBuilder.append(" }");
    }
}

File: org.palladiosimulator.blockchainsystems.core/src/main/kotlin/org/palladiosimulator/blockchainsystems/core/transaction/propagation/TransactionReceivedTraceEvent.kt
package org.palladiosimulator.blockchainsystems.core.transaction.propagation

import org.palladiosimulator.blockchainsystems.core.common.abstractions.TraceEvent
import org.palladiosimulator.blockchainsystems.core.system.abstractions.P2PNetworkEndpoint
import org.palladiosimulator.blockchainsystems.core.transaction.abstractions.Transaction

/**
 * @author Davis Riedel
 */
data class TransactionReceivedTraceEvent(
  override val occurrenceTime: Long,
  val sentTransaction: Transaction,
  val receivingNetworkEndpoint: P2PNetworkEndpoint
) : TraceEvent {
  companion object {
    const val EVENT_TYPE: String = "TransactionReceivedTraceEvent"
  }

  override val eventType = EVENT_TYPE

  override fun formatDetails(stringBuilder: StringBuilder) {
    stringBuilder.append("{ txId=")
    stringBuilder.append(sentTransaction.txId)
    stringBuilder.append("; receivedFromNetworkEndpoint=")
    stringBuilder.append(receivingNetworkEndpoint.getEndpointId())
    stringBuilder.append(" }")
  }
}